### DISCUSSION TITLE
Alignment simplification?

### DISCUSSION THREAD

**Discussion by RVPE** [microsoft/microsoft-ui-xaml#9780] [Ideas]
I would have liked to be able to manage the alignment at the application level :
- Align the left of control foo1 with the left of TextBox foo2 which is inside the control foo3.
- Position the UserControl foo4 so that foo5 in foo4 is centered relative to foo6 (wherever it is).

For example :
Create a TreeView with a TextBlock and a TextBox in the items.
Put the TreeView in a Grid with 3 columns with a GridSplitter in column 1 and align the TexBlock to the column 0 and the TextBox to the column 2.
I don't like spending time on this kind of thing and not getting there in the end.

Maybe instead of having the alignment information in each FrameworkElement, have it in some sort of AlignmentContainer?

I wanted to share this idea, although I don't know if it is good and feasible.

**Comment by NicholasChrzan** [comment]
You might want to look at using RelativePanel it allows to align various controls relative to each other inside the panel.
